### PR TITLE
Unit tests for Utils

### PR DESCRIPTION
**Pull request by hayesjordan**

### CODE REVIEW

**Comment by hayesjordan** [comment]
Empty tests are there to discuss how or if to test those functions.

File: android/PhysicalWeb/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -56,13 +56,10 @@
         <item quantity="one">Beacon Nearby</item>
         <item quantity="other">Beacons Nearby</item>
     </plurals>
-    <string name="title_activity_oob">OobActivity</string>
     <string name="action_settings">Settings</string>
     <string name="oob_accept_and_continue">ACCEPT &amp; CONTINUE</string>
-    <string name="main_prefs_key">physical_web_preferences</string>
     <!-- The following is a more appropriate name, but would require special code to migrate users -->
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/res/values/strings.xml]: you can remove these too I think

**Comment by cco3** [comment]
Please edit travis/build.sh

**Comment by cco3** [comment]
I'll look at the stubs tomorrow

File: android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,634 @@
...
+    Assert.assertTrue(Utils.containsFavorite(pairs));
+  }
+
+  @Test
+  public void testCheckIfUserHasOptedIn() throws Exception {
+
+  }
+
+  @Test
+  public void testDeleteCache() throws Exception {
```
Comment (cco3) [inline android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java]: Yea, this works... you could do something to see if it's empty.

File: android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,634 @@
...
+
+    Mockito.when(mockResources.getIdentifier(GOOGLE_API_KEY_RESOURCE_KEY,
+        GOOGLE_API_KEY_RESOURCE_TYPE, packageName)).thenReturn(resId);
+    Mockito.when(mockContext.getString(resId)).thenReturn(apiKey);
+    Assert.assertEquals(GOOGLE_PWS_URL + "\0" + GOOGLE_PWS_VERSION + "\0" + apiKey,
+        Utils.getDefaultPwsEndpointPreferenceString(mockContext));
+  }
+
+  @Test
+  public void testSetOptInPreference() throws Exception {
```
Comment (cco3) [inline android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java]: I don't think we should bother with this at the moment.

File: android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,634 @@
...
+  public void testSetSharedPreferencesDefaultValues() throws Exception {
+
+  }
+
+  @Test
+  public void testSetPwsEndpointPwsClient() throws Exception {
+  }
+
+  @Test
+  public void testSetPwsEndpointPhysicalWebCollection() throws Exception {
```
Comment (hayesjordan) [inline android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java]: There is a Utils method that calls the set endpoint on the physical web collection.

File: android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,634 @@
...
+        Utils.getDefaultPwsEndpointPreferenceString(mockContext));
+  }
+
+  @Test
+  public void testSetOptInPreference() throws Exception {
+
+  }
+
+  @Test
+  public void testSetPwsEndpointPreference() throws Exception {
```
Comment (cco3) [inline android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java]: These three look fine

File: android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,634 @@
...
+  public void testWarnUserOnMissingApiKey() throws Exception {
+
+  }
+
+  @Test
+  public void testCreateNavigateToUrlIntent() throws Exception {
+  }
+
+  @Test
+  public void testCreateNavigateToUrlPendingIntent() throws Exception {
```
Comment (cco3) [inline android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java]: just a utility, don't bother

File: android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java [hunk 7]
```diff
@@ -0,0 +1,634 @@
...
+
+  }
+
+  @Test
+  public void testWarnUserOnMissingApiKey() throws Exception {
+
+  }
+
+  @Test
+  public void testCreateNavigateToUrlIntent() throws Exception {
```
Comment (cco3) [inline android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java]: Don't bother

File: android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java [hunk 8]
```diff
@@ -0,0 +1,634 @@
...
+  public void testGetMdnsEnabled() throws Exception {
+  }
+
+  @Test
+  public void testGetDebugViewEnabled() throws Exception {
+
+  }
+
+  @Test
+  public void testWarnUserOnMissingApiKey() throws Exception {
```
Comment (cco3) [inline android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java]: Don't bother

File: android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java [hunk 9]
```diff
@@ -0,0 +1,634 @@
...
+  public void testGetCustomPwsEndpoint() throws Exception {
+
+  }
+
+  @Test
+  public void testGetMdnsEnabled() throws Exception {
+  }
+
+  @Test
+  public void testGetDebugViewEnabled() throws Exception {
```
Comment (cco3) [inline android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java]: don't bother

File: android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java [hunk 10]
```diff
@@ -0,0 +1,634 @@
...
+    Assert.assertEquals("", Utils.getGoogleApiKey(context));
+  }
+
+  @Test
+  public void testGetCustomPwsEndpoint() throws Exception {
+
+  }
+
+  @Test
+  public void testGetMdnsEnabled() throws Exception {
```
Comment (cco3) [inline android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java]: don't bother

File: android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java [hunk 11]
```diff
@@ -0,0 +1,634 @@
...
+
+    // Test empty groupId and use site url that causes URISyntaxException
+    url = "\\invalid";
+    Mockito.when(mockResult.getSiteUrl()).thenReturn(url);
+    Mockito.when(mockResult.getGroupId()).thenReturn("");
+    Assert.assertEquals(url, Utils.getGroupId(mockResult));
+  }
+
+  @Test
+  public void testUpdateRegion() throws Exception {
```
Comment (cco3) [inline android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java]: Given that these all just call out to another class, don't bother.

**Comment by hayesjordan** [comment]
@cco3 can you take a quick peek at this again?

**Comment by cco3** [comment]
LGTM

**Pull request merged**